javaslang-benchmark/src/test/java/javaslang/idiom/ForBenchmark.java

```diff
@@ -0,0 +1,122 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import javaslang.collection.Iterator;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static javaslang.API.For;
+import static javaslang.JmhRunner.getRandomValues;
+
+public class ForBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            ForComprehension1Iterator.class,
+            ForComprehension2Iterators.class,
+            ForComprehension3Iterators.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+    @State(Scope.Benchmark)
+    public static class Base {
+        @Param({ "10", "100", "1000" })
+        public int CONTAINER_SIZE;
+
+        Integer[] ELEMENTS;
+        List<Integer> ELEMENTS_ARRAY;
+
+        @Setup
+        public void setup() {
+            ELEMENTS = getRandomValues(CONTAINER_SIZE, 0);
+            ELEMENTS_ARRAY = Arrays.asList(ELEMENTS);
+        }
+    }
+
+    public static class ForComprehension1Iterator extends Base {
+
+        @Benchmark
+        public Object javaslang_for() {
+            final Iterable<Integer> result = For(ELEMENTS_ARRAY)
+                    .yield(i -> i * 0);
+
+            assert Iterator.ofAll(result).forAll(e -> e == 0);
+            return result;
+        }
+
+        @Benchmark
+        public Object java_for() {
+            final List<Integer> result = new ArrayList<>(ELEMENTS_ARRAY.size());
+            for (Integer i : ELEMENTS_ARRAY) {
+                result.add(i * 0);
```

Comment:
This is what I saw:
```
      45: astore_3
      46: aload_1
      47: aload_3
      48: invokevirtual #19                 // Method java/lang/Integer.intValue:()I
      51: iconst_0
      52: imul
      53: invokestatic  #20                 // Method java/lang/Integer.valueOf:(I)Ljava/lang/Integer;
      56: invokeinterface #21,  2           // InterfaceMethod java/util/List.add:(Ljava/lang/Object;)Z
```

I don't know how to check if the mult is elided, but like I said please feel free to point me in the right direction.